compiler/mir/mirgen.nim

```diff
@@ -1094,35 +1064,30 @@ proc genMagic(c: var TCtx, n: PNode; m: TMagic): EValue =
 
   # special macro related magics:
   of mExpandToAst:
-    if n[0].typ == nil:
-      # has no type information -- we need to manually tranlate it
+    # the transformation pass already flattened the call expression for us and
+    # made it a bit easier to process
+    let callee = n[1] # the meta-routine to evaluate
+    case callee.sym.kind
+    of skTemplate:
+      # a ``getAst`` call taking a template call expression. The arguments
+      # need special handling, but the shape stays as is
       argBlock(c.stmts):
-        for i in 1..<n.len:
-          assert n[i].typ == nil
-          # even the argument expressions don't have types in this case, so we
-          # can't use ``argExpr``
-          # FIXME: this happens in the context of ``quote`` (i.e.
-          #        ``semQuoteAst``), because the quoted expressions are not
-          #        semantically analysed.
-          #        The arguments (quoted expressions) should be analysed and
-          #        the call should also use a complete symbol.
```

Comment:
The comment was and is incorrect. The quoted expressions are run through semantic analysis -- only the `getAst` symbol has no type information, as it is created via `createMagic` in `semQuoteAst`.